Logging middleware crashes if response has no data

I can get errors like this in my testing:

```
test/test_framework.py:47:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../.venv/ingress/lib/python3.6/site-packages/hug/test.py:52: in call
    body=body, scheme=scheme), response)
../../.venv/ingress/lib/python3.6/site-packages/falcon/api.py:243: in __call__
    process_response(req, resp, resource, req_succeeded)
../../.venv/ingress/lib/python3.6/site-packages/falcon/api_helpers.py:93: in shim
    process_response(req, resp, resource)
../../.venv/ingress/lib/python3.6/site-packages/hug/middleware.py:103: in process_response
    self.logger.info(self._generate_combined_log(request, response))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sanoingressapi.api.HttpLogger object at 0x1092aab48>, request = <falcon.request.Request object at 0x109257608>, response = <falcon.response.Response object at 0x1093c9de0>

    def _generate_combined_log(self, request, response):
        """Given a request/response pair, generate a logging format similar to the NGINX combined style."""
        current_time = datetime.utcnow()
        return '{0} - - [{1}] {2} {3} {4} {5} {6}'.format(request.remote_addr, current_time, request.method,
                                                        request.relative_uri, response.status,
>                                                       len(response.data), request.user_agent)
E       TypeError: object of type 'NoneType' has no len()

../../.venv/ingress/lib/python3.6/site-packages/hug/middleware.py:95: TypeError
```

PR forthcoming.

--- a/hug/middleware.py
+++ b/hug/middleware.py
@@ -90,9 +90,10 @@ class LogMiddleware(object):
     def _generate_combined_log(self, request, response):
         """Given a request/response pair, generate a logging format similar to the NGINX combined style."""
         current_time = datetime.utcnow()
+        data_len = '-' if response.data is None else len(response.data)
         return '{0} - - [{1}] {2} {3} {4} {5} {6}'.format(request.remote_addr, current_time, request.method,
                                                         request.relative_uri, response.status,
-                                                        len(response.data), request.user_agent)
+                                                        data_len, request.user_agent)
 
     def process_request(self, request, response):
         """Logs the basic endpoint requested"""

